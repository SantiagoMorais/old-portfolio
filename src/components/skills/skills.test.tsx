import { render, screen } from "@testing-library/react";
import { Skills } from ".";

describe("<Skills />", () => {
  it("should render the component correctly", () => {
    render(<Skills />);
    const title = screen.getByText(/skills/i);
    expect(title).toBeInTheDocument();
  });

  it("should render all skills categories", () => {
    render(<Skills />);
    const skills = document.querySelectorAll(".skill");
    expect(skills).toHaveLength(4);
  });

  it("should render the list skills correctly", () => {
    render(<Skills />);
    const skillLists = document.querySelector(".skillLists");
    expect(skillLists).toBeInTheDocument();
    expect(skillLists).matchSnapshot();
  });
});
