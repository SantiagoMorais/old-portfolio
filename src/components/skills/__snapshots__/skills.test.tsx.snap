// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Skills /> > should render the list skills correctly 1`] = `
<div
  class="skillLists"
>
  <div
    class="skill"
  >
    <div
      class="subtitle"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-code icon"
        data-icon="code"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M392.8 1.2c-17-4.9-34.7 5-39.6 22l-128 448c-4.9 17 5 34.7 22 39.6s34.7-5 39.6-22l128-448c4.9-17-5-34.7-22-39.6zm80.6 120.1c-12.5 12.5-12.5 32.8 0 45.3L562.7 256l-89.4 89.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l112-112c12.5-12.5 12.5-32.8 0-45.3l-112-112c-12.5-12.5-32.8-12.5-45.3 0zm-306.7 0c-12.5-12.5-32.8-12.5-45.3 0l-112 112c-12.5 12.5-12.5 32.8 0 45.3l112 112c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256l89.4-89.4c12.5-12.5 12.5-32.8 0-45.3z"
          fill="currentColor"
        />
      </svg>
      <h3
        class="skillTitle"
      >
        Technical Proficiencies
      </h3>
    </div>
    <ul
      class="list"
    >
      <li
        class="item"
      >
        React JS
      </li>
      <li
        class="item"
      >
        JavaScript
      </li>
      <li
        class="item"
      >
        TypeScript
      </li>
      <li
        class="item"
      >
        HTML5
      </li>
      <li
        class="item"
      >
        CSS3
      </li>
      <li
        class="item"
      >
        Git
      </li>
      <li
        class="item"
      >
        Redux
      </li>
      <li
        class="item"
      >
        Context API
      </li>
      <li
        class="item"
      >
        Jest
      </li>
      <li
        class="item"
      >
        Testing Library
      </li>
      <li
        class="item"
      >
        Vitest
      </li>
    </ul>
  </div>
  <div
    class="skill"
  >
    <div
      class="subtitle"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-screwdriver-wrench icon"
        data-icon="screwdriver-wrench"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M78.6 5C69.1-2.4 55.6-1.5 47 7L7 47c-8.5 8.5-9.4 22-2.1 31.6l80 104c4.5 5.9 11.6 9.4 19 9.4h54.1l109 109c-14.7 29-10 65.4 14.3 89.6l112 112c12.5 12.5 32.8 12.5 45.3 0l64-64c12.5-12.5 12.5-32.8 0-45.3l-112-112c-24.2-24.2-60.6-29-89.6-14.3l-109-109V104c0-7.5-3.5-14.5-9.4-19L78.6 5zM19.9 396.1C7.2 408.8 0 426.1 0 444.1C0 481.6 30.4 512 67.9 512c18 0 35.3-7.2 48-19.9L233.7 374.3c-7.8-20.9-9-43.6-3.6-65.1l-61.7-61.7L19.9 396.1zM512 144c0-10.5-1.1-20.7-3.2-30.5c-2.4-11.2-16.1-14.1-24.2-6l-63.9 63.9c-3 3-7.1 4.7-11.3 4.7H352c-8.8 0-16-7.2-16-16V102.6c0-4.2 1.7-8.3 4.7-11.3l63.9-63.9c8.1-8.1 5.2-21.8-6-24.2C388.7 1.1 378.5 0 368 0C288.5 0 224 64.5 224 144l0 .8 85.3 85.3c36-9.1 75.8 .5 104 28.7L429 274.5c49-23 83-72.8 83-130.5zM56 432a24 24 0 1 1 48 0 24 24 0 1 1 -48 0z"
          fill="currentColor"
        />
      </svg>
      <h3
        class="skillTitle"
      >
        Tools
      </h3>
    </div>
    <ul
      class="list"
    >
      <li
        class="item"
      >
        Visual Studio Code
      </li>
      <li
        class="item"
      >
        GitHub
      </li>
      <li
        class="item"
      >
        Git
      </li>
      <li
        class="item"
      >
        Figma
      </li>
    </ul>
  </div>
  <div
    class="skill"
  >
    <div
      class="subtitle"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-book icon"
        data-icon="book"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 0C43 0 0 43 0 96V416c0 53 43 96 96 96H384h32c17.7 0 32-14.3 32-32s-14.3-32-32-32V384c17.7 0 32-14.3 32-32V32c0-17.7-14.3-32-32-32H384 96zm0 384H352v64H96c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16H336c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16zm16 48H336c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
          fill="currentColor"
        />
      </svg>
      <h3
        class="skillTitle"
      >
        React Libraries
      </h3>
    </div>
    <ul
      class="list"
    >
      <li
        class="item"
      >
        Styled-Components
      </li>
      <li
        class="item"
      >
        Axios
      </li>
      <li
        class="item"
      >
        React-Query
      </li>
      <li
        class="item"
      >
        React-Hook-Form
      </li>
      <li
        class="item"
      >
        React-Router-Dom
      </li>
    </ul>
  </div>
  <div
    class="skill"
  >
    <div
      class="subtitle"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-key icon"
        data-icon="key"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M336 352c97.2 0 176-78.8 176-176S433.2 0 336 0S160 78.8 160 176c0 18.7 2.9 36.8 8.3 53.7L7 391c-4.5 4.5-7 10.6-7 17v80c0 13.3 10.7 24 24 24h80c13.3 0 24-10.7 24-24V448h40c13.3 0 24-10.7 24-24V384h40c6.4 0 12.5-2.5 17-7l33.3-33.3c16.9 5.4 35 8.3 53.7 8.3zM376 96a40 40 0 1 1 0 80 40 40 0 1 1 0-80z"
          fill="currentColor"
        />
      </svg>
      <h3
        class="skillTitle"
      >
        Key Attributes
      </h3>
    </div>
    <ul
      class="list"
    >
      <li
        class="item"
      >
        Organized
      </li>
      <li
        class="item"
      >
        Great Receiving Feedbacks
      </li>
      <li
        class="item"
      >
        Proactive Collaboration
      </li>
      <li
        class="item"
      >
        Commitment
      </li>
      <li
        class="item"
      >
        Adaptability
      </li>
      <li
        class="item"
      >
        Resilience
      </li>
      <li
        class="item"
      >
        Attention to the Details
      </li>
    </ul>
  </div>
</div>
`;
